packages/core/src/NodeExecuteFunctions.ts

```diff
@@ -2508,6 +2508,19 @@ const getCommonWorkflowFunctions = (
 	prepareOutputData: async (outputData) => [outputData],
 });
 
+const executionCancellationFunctions = (
+	abortSignal: AbortSignal,
+): Pick<IExecuteFunctions, 'onExecutionCancellation' | 'getExecutionCancelSignal'> => ({
+	getExecutionCancelSignal: () => abortSignal,
+	onExecutionCancellation: (handler) => {
+		const fn = () => {
+			abortSignal.removeEventListener('abort', fn);
+			handler();
+		};
```

Comment:
Would it be useful to log here that a cancellation occurred?

Reply:
Do you mean logging for every node being aborted? or do you mean we should log when an execution is aborted.
This code is called for individual node that wishes to do something on execution cancellation.

Reply:
Exactly, I meant for every node aborted - but on second thought, it might be enough to log when the execution itself is cancelled.

Reply:
for that to work, we'd have to add the log to `onCancel`, while all the other execution finishing logs are in `processSuccessExecution`. After `PCancelable` is gone, this will be less mess, and I'll add this logging then.